itest/lnd_psbt_test.go

```diff
@@ -1340,3 +1341,51 @@ func sendAllCoinsToAddrType(ht *lntest.HarnessTest,
 	ht.MineBlocksAndAssertNumTxes(1, 1)
 	ht.WaitForBlockchainSync(hn)
 }
+
+// testPsbtChanFunfingFailFlow tests the failing of a funding flow by the
+// remote peer and that the initiator receives the expected error and aborts
+// the channel opening. The psbt funding flow is used to simulate this behavior
+// because we can easily let the remote peer run into the timeout.
+func testPsbtChanFunfingFailFlow(ht *lntest.HarnessTest) {
+	const chanSize = funding.MaxBtcFundingAmount
+
+	// Decrease the timeout window for the remote peer to accelerate the
+	// funding fail process.
+	args := []string{
+		"--reservationtimeout=1s",
+		"--zombiesweeperinterval=1s",
+	}
+	ht.RestartNodeWithExtraArgs(ht.Bob, args)
+
+	// Before we start the test, we'll ensure both sides are connected so
+	// the funding flow can be properly executed.
+	alice := ht.Alice
+	bob := ht.Bob
+	ht.EnsureConnected(alice, bob)
+
+	// At this point, we can begin our PSBT channel funding workflow. We'll
+	// start by generating a pending channel ID externally that will be used
+	// to track this new funding type.
+	pendingChanID := ht.Random32Bytes()
+
+	// Now that we have the pending channel ID, Alice will open the channel
+	// by specifying a PSBT shim.
+	chanUpdates, _ := ht.OpenChannelPsbt(
+		alice, bob, lntest.OpenChannelParams{
+			Amt: chanSize,
+			FundingShim: &lnrpc.FundingShim{
+				Shim: &lnrpc.FundingShim_PsbtShim{
+					PsbtShim: &lnrpc.PsbtShim{
+						PendingChanId: pendingChanID,
+					},
+				},
+			},
+		},
+	)
+
+	// We received the AcceptChannel msg from our peer but we are not going
+	// to fund this channel but instead wait for our peer to fail the
+	// funding workflow with an internal error.
+	expectedErr := fmt.Errorf("funding failed due to internal error")
```

Comment:
should we maybe more more specific and match on `ErrRemoteCanceled`?

Reply:
hmm so we as the remote here do not fail with `ErrRemoteCanceled` but with the generic internal error.

Reply:
when i print out the error, I get: 
```
rpc error: code = Unknown desc = remote canceled funding, possibly timed out: received funding error from 022113e2611f7a60ecaba03123e014800017727c8e4914e174934b8802cb662a5d: chan_id=c03aa5ef88a320f9b91f97f8d5273ffa8e3d7c0327c6ec25d4f3f8f56283e4b5, err=funding failed due to internal error
```

Reply:
https://github.com/lightningnetwork/lnd/blob/master/funding/manager.go#L940

Reply:
Ahh ok, you right a lot of wrapped errors lol, will go with your recommendation.